miner/worker.go

```diff
@@ -192,8 +193,11 @@ func (w *worker) commitNewWork() (*types.Block, error) {
 		return nil, err
 	}
 
-	// Fill the block with all available pending transactions.
+	// Get the pending txs from TxPool
 	pending := w.eth.TxPool().Pending(true)
+	// Filter out transactions that don't satisfy predicateContext and remove them from TxPool
+	rules := w.chainConfig.AvalancheRules(header.Number, new(big.Int).SetUint64(header.Time))
+	pending = w.enforcePredicates(rules, predicateContext, pending)
```

Comment:
should we add a nil check on predicateContext here to avoid needing to reason about the behavior of other functions with a nil predicateContext?

Reply:
It's valid for the engine to leave the predicateContext nil, so it's up to the active precompiles to determine whether or not a nil predicateContext is valid.

Reply:
Updated to separate out two different contexts one with the ProposerVM wrapper so that it can be set only if needed by the ProposerVM, and one that does not require it so that it can be used within shared memory without impacting the optional interfaces `ShouldVerifyWithContext` and `VerifyWithContext`